### PR TITLE
Add shellcheck to git hooks and convert to xtask.

### PR DESCRIPTION
**Pull request by Alexhuszagh**
Convert git hooks to use `xtask` commands, and add `shellcheck` to `xtask check`. Can run on staged or all files.

### CODE REVIEW

File: xtask/src/hooks.rs [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+fn is_shell_script(path: impl AsRef<Path>) -> cross::Result<bool> {
+    if path.as_ref().is_dir() {
+        // is a directory if a git submodule
+        return Ok(false);
+    }
+    let file = File::open(path.as_ref())?;
+    let reader = BufReader::new(file);
+
+    match reader.lines().next() {
+        Some(Ok(line)) => Ok(line.starts_with("#!") && line.contains("sh")),
```
Comment (Alexhuszagh) [inline xtask/src/hooks.rs:110]: A more sophisticated way would be to actually evaluate the shebang line, but this is identical to what we did before, and it's going to work unless we **really** try to mess it up, so I think this is fine.

File: xtask/src/hooks.rs [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    verbose: bool,
+}
+
+fn has_nightly(verbose: bool) -> cross::Result<bool> {
+    Command::new("cargo")
+        .arg("+nightly")
+        .run_and_get_output(verbose)
+        .map(|o| o.status.success())
+        .map_err(Into::into)
+}
```
Comment (Emilgardis) [inline xtask/src/hooks.rs:34]: technically, this should be
```suggestion
fn has_nightly(verbose: bool) -> cross::Result<bool> {
    Command::new("cargo")
        .arg("+nightly")
        .run_and_get_output(verbose)
        .map(|o| o.stdout.contains("-nightly"))
        .map_err(Into::into)
}
```

but that's just being very pedantic

**Review by Emilgardis — APPROVED** [review]

File: xtask/src/hooks.rs [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+                .args(&scripts)
+                .run(verbose, false)?;
+        }
+    }
+
+    Ok(())
+}
+
+pub fn check(Check { verbose, all }: Check, toolchain: Option<&str>) -> cross::Result<()> {
+    println!("Running rustfmt and clippy checks.");
```
Comment (Emilgardis) [inline xtask/src/hooks.rs:143]: missing shellsheck

File: xtask/src/hooks.rs (hunk 2 above)
Comment (Alexhuszagh) [reply xtask/src/hooks.rs:34]: Why though? I've just checked that running `cargo +nightly` produces a non-zero error code: if it works, it works. If someone wants to rename it something funny, then sure. If I do:

```bash
$ cargo +sofa
error: toolchain 'sofa' is not installed
```

Comment (Emilgardis) [reply xtask/src/hooks.rs:34]: exactly, it's very uneccessary :)

**Comment by Alexhuszagh** [comment]
bors r=Emilgardis

**Pull request merged**